From eac7cdabecafb7c505795207182ab2578d672c06 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Tue, 16 Mar 2010 11:29:08 -0700
Subject: [PATCH 01/37] Revert "XQuartz: Explicitly pass a bellProc to make XBell() work again."

I'm not quite sure why this was necessary, but DDXRingBell is being called
from CoreKeyboardBell, so we don't need a separate bellProc which would
result in multiple rings.

This reverts commit 9071b0d69748cfa7ecca17b4cb0e431bbb0ef2a4.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 hw/xquartz/quartzKeyboard.c |    8 +-------
 1 files changed, 1 insertions(+), 7 deletions(-)

diff --git xorg-server-1.7.99.902/hw/xquartz/quartzKeyboard.c xorg-server-1.7.99.902/hw/xquartz/quartzKeyboard.c
index 7e36a9a..c9ef7cc 100644
--- xorg-server-1.7.99.902/hw/xquartz/quartzKeyboard.c
+++ xorg-server-1.7.99.902/hw/xquartz/quartzKeyboard.c
@@ -184,12 +184,6 @@ static void DarwinChangeKeyboardControl(DeviceIntPtr device, KeybdCtrl *ctrl) {
     // keyclick, bell volume / pitch, autorepead, LED's
 }
 
-static void DarwinKeyboardBell(int volume, DeviceIntPtr pDev, pointer arg, int something) {
-    KeybdCtrl *ctrl = arg;
-
-    DDXRingBell(volume, ctrl->bell_pitch, ctrl->bell_duration);
-}
-
 //-----------------------------------------------------------------------------
 // Utility functions to help parse Darwin keymap
 //-----------------------------------------------------------------------------
@@ -301,7 +295,7 @@ void DarwinKeyboardInit(DeviceIntPtr pDev) {
     // for a kIOHIDParamConnectType connection.
     assert(darwinParamConnect = NXOpenEventStatus());
 
-    InitKeyboardDeviceStruct(pDev, NULL, DarwinKeyboardBell, DarwinChangeKeyboardControl);
+    InitKeyboardDeviceStruct(pDev, NULL, NULL, DarwinChangeKeyboardControl);
 
     DarwinKeyboardReloadHandler();
 
-- 
1.7.0.2

